src/main/kotlin/net/maizegenetics/phgv2/cli/AgcCompress.kt

```diff
@@ -0,0 +1,228 @@
+        try {
+            var process = builder.start()
+            var error = process.waitFor()
+            if (error != 0) {
+                println("agc create run via ProcessBuilder returned error code $error")
+                throw IllegalArgumentException("Error running ProcessBuilder to compress agc files: ${error}")
+            }
+            // If this was an "append" command, the file was written to assemblies_tmp.agc
+            // In this case, we need to move the original assemblies.agc file to assemblies_backup_<date>.agc
+            // can then move the assemblies_tmp.agc to assemblies.agc
+
+            if (loadOption == "append") {
+                // Use Files.move instead of File.renameTo because File.renameTo does not work across file systems
+                // and Files.move is platform independent
+                val agcFileBackupFile = Paths.get("${dbPath}/assemblies_backup_${LocalDate.now()}.agc")
+                val agcFileOutFile = Paths.get(agcFileOut)
+
+                val agcFileOrig = Paths.get(agcFile)
+                Files.move(agcFileOrig,agcFileBackupFile)
+                Files.move(agcFileOutFile,agcFileOrig)
+            }
+
+        } catch (exc: Exception) {
+            myLogger.error("Error: could not create agc compressed file.")
+            throw IllegalArgumentException("Error running ProcessBuilder for agc create or append: ${exc.message}")
+        }
+
+        return true
+    }
+
+    // This function takes a list of fasta file names and returns a list of sample names
+    // Sample names are the file names without extension
+    fun getCurrentSampleNames(fastaFiles:List<String>):List<String> {
+        //create a list of names from the fasta files, where the name is
+        // just the file name, no path, and without the extension
+        val sampleNames = mutableListOf<String>()
+        fastaFiles.forEach { sampleNames.add(it.split("/").last().split(".").first()) }
+        return sampleNames
+    }
+
+    fun getSampleListFromAGC(agcFile:String,tempDir:String): List<String> {
+        // This function will return a list of samples from the AGC compressed file.
+        // This will be used to verify that the new list of fastas has nothing overlapping
+        // the exsiting fastas in the AGC compressed file.
+
+        var sampleList = listOf<String>()
+        // Query the agc compressed file to get list of sample names
+        var builder = ProcessBuilder("conda","run","-n","phgv2-conda","agc","listset",agcFile)
+        var redirectOutput = tempDir + "/agc_create_output.log"
+        var redirectError = tempDir + "/agc_create_error.log"
+        builder.redirectOutput( File(redirectOutput))
+        builder.redirectError( File(redirectError))
+
+        myLogger.info("begin Command to list agc sample names:" + builder.command().stream().collect(Collectors.joining(" ")))
```

Comment:
get rid of .stream().  Use kotlin collections/methods.

Reply:
done